### PR TITLE
Added support for window icon

### PR DESCRIPTION
**Pull request by Memorix101**
Added icon support in WindowsGameForm.cs :)
The osu!.res file from osu.Desktop is used for that.
So, the the icon is stored only once ;)

### CODE REVIEW

**Comment by tsudoko** [comment]
The diff is unreadable because all line endings in source files have been changed from `\n` to `\r\n`. You'll probably want to fix that.

**Comment by Memorix101** [comment]
Sorry, MonoDevelop did that :(

File: osu.Framework.Desktop/OS/DesktopGameWindow.cs [hunk 1]
```diff
@@ -41,3 +42,48 @@ protected override void SetTitle(string title)
...
+            Form.Close();
+        }
+
+        protected override void SetTitle(string title)
+        {
+            Form.Text = title;
+        }
+    }
+}
+>>>>>>> 95f68465bc8a33a9bab2caa996457390b2326b88
```
Comment (smoogipooo) [inline osu.Framework.Desktop/OS/DesktopGameWindow.cs]: Fix the merge conflict up.

File: osu.Framework.Desktop/OS/Windows/WindowsGameForm.cs [hunk 2]
```diff
@@ -31,6 +31,9 @@ internal WindowsGameForm(GraphicsContextFlags flags) : base(flags)
 
             LocationChanged += delegate { updateScreen(); };
             ClientSizeChanged += delegate { updateScreen(); };
+
+			this.Icon = Icon.ExtractAssociatedIcon(System.Reflection.Assembly.GetExecutingAssembly().Location);
+
```
Comment (smoogipooo) [inline osu.Framework.Desktop/OS/Windows/WindowsGameForm.cs]: Your editor is using hard tabs to indent. We use soft tabs.

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj [hunk 3]
```diff
@@ -18,9 +18,9 @@
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;WINDOWS</DefineConstants>
```
Comment (smoogipooo) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: What's this WINDOWS constant?

Comment (Memorix101) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: I added it for debugging between Unix platforms and Windows. It's not needed any more. I removed it.

Comment (smoogipooo) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: Let's not use conditional compilation constants period.

File: osu.Framework.Desktop/OS/DesktopGameWindow.cs [hunk 4]
```diff
@@ -41,3 +42,48 @@ protected override void SetTitle(string title)
         }
     }
 }
+=======
```
Comment (peppy) [inline osu.Framework.Desktop/OS/DesktopGameWindow.cs]: seriously? what is this?

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj [hunk 5]
```diff
@@ -18,9 +18,9 @@
     <DebugType>full</DebugType>
     <Optimize>false</Optimize>
     <OutputPath>bin\Debug\</OutputPath>
-    <DefineConstants>DEBUG;TRACE</DefineConstants>
+    <DefineConstants>TRACE;DEBUG;WINDOWS</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
-    <WarningLevel>4</WarningLevel>
+    <WarningLevel>3</WarningLevel>
```
Comment (peppy) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: why is this changed

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj [hunk 6]
```diff
@@ -30,6 +30,13 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
+  <PropertyGroup>
+      <Win32Resource>../../osu.Desktop/osu!.res</Win32Resource>
```
Comment (peppy) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: no

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj [hunk 7]
```diff
@@ -101,6 +108,7 @@
     <None Include="packages.config" />
   </ItemGroup>
   <ItemGroup>
+    <Content Include="lazer.ico" />
```
Comment (peppy) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: no

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj [hunk 8]
```diff
@@ -157,11 +165,11 @@
     </Content>
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <!-- To modify your build process, add your task inside one of the targets below and uncomment it. 
+  <!-- To modify your build process, add your task inside one of the targets below and uncomment it.
```
Comment (peppy) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: ?

**Comment by peppy** [comment]
learn to view your git diffs before pushing or pull requesting. basically every change you've made here is wrong (most because they are unnecessary changes) and could have been fixed if you knew how to use git.

go master git before you try and help out here, please.

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj (hunk 6 above)
Comment (Memorix101) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: this is needed to make it work crossplatform. otherwise you can't compile it with icons enabled.

**Comment by Memorix101** [comment]
ok. i will redone my stuff here. then I try to do it right.

File: osu.Framework.Desktop/osu.Framework.Desktop.csproj (hunk 6 above)
Comment (peppy) [inline osu.Framework.Desktop/osu.Framework.Desktop.csproj]: no

File: osu.Framework.Desktop/OS/DesktopGameWindow.cs (hunk 4 above)
Comment (Memorix101) [inline osu.Framework.Desktop/OS/DesktopGameWindow.cs]: some formatting issue I've had.
